custom-components/react-carousel/carousel.jsx

```diff
@@ -0,0 +1,110 @@
+const newId = (() => {
+  let uid = 0
+  return () => uid++
+})()
+
+function Product(props) {
+  const {Deeplink, ImageURL, Price, Title} = props.record
+  return (
+    <a href={Deeplink} className="product">
+      <img src={ImageURL} alt={Title} />
+      <div className="title">{Title}</div>
+      <div className="price">{Price}</div>
+    </a>
+  )
+}
+
+class Carousel extends React.Component {
+  state = {
+    offset: 0,
+    records: []
+  }
+
+  constructor(props) {
+    super(props)
+    this.uid = newId()
+    this.ref = React.createRef()
+  }
+
+  async componentDidMount() {
+    const {EventAggregator, ResultDispatcher} = (await withFactfinder()).communication
+
+    setTimeout(() => EventAggregator.addFFEvent({
```

Comment:
For the completeness, we should also react to the `query` prop changes. In the old version with hooks, this piece of code was run on every update.